pkg/destroy/aws/elbhelpers.go

```diff
@@ -105,6 +107,18 @@ func deleteElasticLoadBalancerTargetGroup(ctx context.Context, client *elbv2.ELB
 	return nil
 }
 
+func deleteElasticLoadBalancerListener(ctx context.Context, client *elbv2.ELBV2, arn arn.ARN, logger logrus.FieldLogger) error {
+	_, err := client.DeleteListenerWithContext(ctx, &elbv2.DeleteListenerInput{
+		ListenerArn: aws.String(arn.String()),
+	})
+	if err != nil {
+		return err
```

Comment:
The AWS issue is about danging tags after the listener and load balancer have been removed.  So if you go this route, you'll probably want logic like [this][1] to say "ok, I guess it's already gone, and we can ignore this tag going forward".

But that might expose us to a race where the listener actually does exist and has been tagged, but due to some eventual-consistency hiccup we get a not-found response when trying to delete it.  Although in that case, I suppose we'd fail to delete the VPC, and the user could launch a fresh deprovision attempt, and by then maybe the eventual consistency issue would have resolved, and we'd be able to delete the listener.

[1]: https://github.com/openshift/installer/blob/5a9ad4c024091c6658813ab3a04c8deb61d684cc/pkg/destroy/aws/aws.go#L786

Reply:
By that logic, wouldn't the `targetgroup` resource type potentially suffer from the same problem?

Is there another route you'd suggest?

Reply:
Done.